testing/conftest.py

```diff
@@ -13,5 +24,6 @@ def finish():
         while atexit_fns:
             atexit_fns.pop()()
 
-    monkeypatch.setattr(atexit, "register", atexit_fns.append)
+    monkeypatch.setattr(atexit, "register", lambda func, *args, **kwargs:
+                        atexit_fns.append(AtExitFunc(func, *args, **kwargs)))
```

Comment:
Can't this be just
```suggestion
                        atexit_fns.append(functools.partial(func, *args, **kwargs)))
```
? w/o adding an extra class?

Reply:
Err, good point, that should work. Didn't think of that!

Reply:
Updated, but with a slightly different solution.